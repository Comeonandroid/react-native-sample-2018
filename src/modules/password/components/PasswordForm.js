import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

import Yup from '../../../utils/yup'
import Icon from '../../../utils/icon'
import { t } from '../../../locales/i18n'

import { SubmitContainer, } from '../../../elements'
import { IconTextfield } from '../../../elements'
import { SubmitButton } from '../../../elements'
import { BaseForm, FormContainer, FormGroup, ValidationError } from '../../../elements'

export default class PasswordForm extends PureComponent {

  constructor(props) {
    super(props)

    this.validationSchema = this.validationSchema.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
    this.renderForm = this.renderForm.bind(this)
  }

  validationSchema() {
    return Yup.object().shape({
      password: Yup.string().required(t('inputs.errors.required'))
    })
  }

  handleSubmit(values, actions) {
    this.props.onSend(values)
    actions.setSubmitting(false)
  }

  renderForm(props) {
    const {
      values,
      errors,
      touched,
      setFieldValue,
      isSubmitting,
      handleSubmit,
      setFieldTouched
    } = props
    return (
      <FormContainer>
        <FormGroup key={'password'}>
          <IconTextfield
            name={'password'}
            value={values.password}
            error={errors.password && touched.password}
            placeholder={t(`inputs.${'password'}.placeholder`)}
            onBlur={setFieldTouched}
            onChange={setFieldValue}
            type={'password'} >
            <Icon name='password' size={16} />
          </IconTextfield>
          {errors.password && touched.password && <ValidationError>{errors.password}</ValidationError>}
        </FormGroup>
        <SubmitContainer>
          <SubmitButton enabled={!isSubmitting} rounded onPress={handleSubmit}>{t('inputs.submit')}</SubmitButton>
        </SubmitContainer>
      </FormContainer>
    )
  }

  render() {
    return (
      <BaseForm
        initialValues={this.props.values}
        validationSchema={this.validationSchema}
        onSubmit={this.handleSubmit}
        render={this.renderForm}
      />
    )
  }
}

PasswordForm.propTypes = {
  onSend: PropTypes.func.isRequired,
  values: PropTypes.shape({
    password: PropTypes.string
  })
}

PasswordForm.defaultProps = {
  values: {
    password: ''
  }
}