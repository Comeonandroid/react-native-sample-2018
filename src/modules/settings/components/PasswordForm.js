import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

import Yup from '../../../utils/yup'
import { t } from '../../../locales/i18n'

import { SubmitContainer, } from '../../../elements'
import { BaseForm, FormContainer, FormGroup, ValidationError } from '../../../elements'
import { Textfield } from '../../../elements'
import { SubmitButton } from '../../../elements'

export default class PasswordForm extends PureComponent {

  constructor(props) {
    super(props)

    this.validationSchema = this.validationSchema.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
    this.renderForm = this.renderForm.bind(this)
  }

  validationSchema() {
    return Yup.object().shape({
      password: Yup.string().required(t('inputs.errors.required')),
      password_new: Yup.string().required(t('inputs.errors.required')),
      password_repeat: Yup.string().equalTo(Yup.ref('password_new'), t('inputs.errors.match')).required(t('inputs.errors.required'))
    })
  }

  handleSubmit(values, actions) {
    this.props.onSend(values)
    actions.setSubmitting(false)
  }

  renderForm(props) {
    const {
      values,
      errors,
      touched,
      setFieldValue,
      isSubmitting,
      handleSubmit,
      setFieldTouched
    } = props
    return (
      <FormContainer>
          {['password', 'password_new', 'password_repeat'].map( key => (
            <FormGroup key={key}>
              <Textfield
                name={key}
                type='password'
                value={values[key]}
                error={errors[key] && touched[key]}
                placeholder={t(`inputs.${key}.placeholder`)}
                onBlur={setFieldTouched}
                onChange={setFieldValue} />
                { errors[key] && touched[key] && <ValidationError>{errors[key]}</ValidationError> }
            </FormGroup>
          ))}
        <SubmitContainer>
          <SubmitButton enabled={!isSubmitting} rounded onPress={handleSubmit}>{t('inputs.confirm')}</SubmitButton>
        </SubmitContainer>
      </FormContainer>
    )
  }

  render() {
    return (
      <BaseForm
        initialValues={this.props.values}
        validationSchema={this.validationSchema}
        onSubmit={this.handleSubmit}
        render={this.renderForm}
      />
    )
  }
}

PasswordForm.propTypes = {
  onSend: PropTypes.func.isRequired,
  values: PropTypes.shape({
    password: PropTypes.string,
    password_new: PropTypes.string,
    password_repeat: PropTypes.string
  })
}

PasswordForm.defaultProps = {
  values: {
    password: '',
    password_new: '',
    password_repeat: ''
  }
}

