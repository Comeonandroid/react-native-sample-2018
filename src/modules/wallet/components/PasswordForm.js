import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

import Yup from '../../../utils/yup'
import Icon from '../../../utils/icon'
import { t } from '../../../locales/i18n'

import { SubmitContainer, } from '../../../elements'
import { BaseForm, FormContainer, FormGroup, ValidationError } from '../../../elements'
import { IconTextfield } from '../../../elements'
import { SubmitButton } from '../../../elements'
import { WarningLabel } from '../../../elements'

import TermsCheckboxContainer from '../containers/TermsCheckboxContainer'

export default class PasswordForm extends PureComponent {

  constructor(props) {
    super(props)

    this.validationSchema = this.validationSchema.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
    this.renderForm = this.renderForm.bind(this)
    this.handleTermsChange = this.handleTermsChange.bind(this)
  }

  validationSchema() {
    return Yup.object().shape({
      password: Yup.string().required(t('inputs.errors.required')),
      password_repeat: Yup.string().equalTo(Yup.ref('password'), t('inputs.errors.match')).required(t('inputs.errors.required'))
    })
  }

  handleSubmit(values, actions) {
    this.props.onSend(values)
    actions.setSubmitting(false)
  }

  handleTermsChange(event) {
    console.log('event', event)
  }

  renderForm(props) {
    const { reset } = this.props
    const {
      values,
      errors,
      touched,
      setFieldValue,
      isSubmitting,
      handleSubmit,
      setFieldTouched
    } = props
    return (
      <FormContainer>
        {reset && <WarningLabel>{t('wallet.reset_password_aware')}</WarningLabel> }
        {['password', 'password_repeat'].map(key => (
          <FormGroup key={key}>
            <IconTextfield
              name={key}
              type='password'
              value={values[key]}
              error={errors[key] && touched[key]}
              placeholder={t(`inputs.${key}.placeholder`)}
              onBlur={setFieldTouched}
              onChange={setFieldValue}>
              <Icon name='password' size={16} />
            </IconTextfield>
            {errors[key] && touched[key] && <ValidationError>{errors[key]}</ValidationError>}
          </FormGroup>
        ))}
        <TermsCheckboxContainer onCheckboxChange={this.handleTermsChange} />
        <SubmitContainer>
          <SubmitButton enabled={!isSubmitting} rounded onPress={handleSubmit}>{t('inputs.confirm')}</SubmitButton>
        </SubmitContainer>
      </FormContainer>
    )
  }

  render() {
    return (
      <BaseForm
        initialValues={this.props.values}
        validationSchema={this.validationSchema}
        onSubmit={this.handleSubmit}
        render={this.renderForm}
      />
    )
  }
}

PasswordForm.propTypes = {
  reset: PropTypes.bool,
  onSend: PropTypes.func.isRequired,
  values: PropTypes.shape({
    password: PropTypes.string,
    password_repeat: PropTypes.string
  })
}

PasswordForm.defaultProps = {
  values: {
    password: '',
    password_repeat: ''
  }
}
