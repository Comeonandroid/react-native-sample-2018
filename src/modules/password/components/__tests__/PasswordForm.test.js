import 'react-native'
import React from 'react'

import { itBehavesLike } from '../../../../specs/shared_examples/index'

import PasswordForm from '../PasswordForm'

const setup = propOverrides => {
  const props = Object.assign({
    onSend: jest.fn(),
    values: {
      passowrd: ''
    }
  }, propOverrides)

  const component = <PasswordForm {...props} />

  return {
    props,
    component
  }
}

describe('<PasswordForm />', () => {
  const { component } = setup()
  itBehavesLike('aThemedComponent', { subject: component })

  describe('values', () => {
    const { component } = setup({
      values: {
        password: 'testtest'
      }
    })
    itBehavesLike('aThemedComponent', { subject: component })
  })
})
