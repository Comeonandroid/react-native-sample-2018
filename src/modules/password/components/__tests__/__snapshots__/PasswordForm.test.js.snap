// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PasswordForm /> dark: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {
          "paddingBottom": 0,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 0,
        },
        undefined,
      ],
    ]
  }
>
  <View
    bottom={undefined}
    style={
      Array [
        Object {},
        Array [
          Object {
            "marginBottom": 0,
            "marginTop": 15,
          },
          undefined,
        ],
      ]
    }
    top={undefined}
  >
    <View
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "left": 9,
              "position": "absolute",
              "top": 7,
              "width": 40,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#c1c1c1",
                "fontSize": 16,
              },
              Array [
                Object {
                  "color": "#c1c1c1",
                },
                undefined,
              ],
              Object {
                "fontFamily": "icomoon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "height": 39,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onBlur={[Function]}
          onChange={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          password={true}
          placeholder="Type your password.."
          placeholderTextColor="#eeeeee"
          secureTextEntry={true}
          selectionColor="#faa632"
          style={
            Array [
              Object {
                "alignSelf": "stretch",
                "backgroundColor": "transparent",
                "marginTop": 0,
                "paddingBottom": 2,
                "paddingTop": 2,
              },
              Object {
                "color": "rgb(117, 117, 117)",
                "fontSize": 16,
                "paddingLeft": 0,
                "paddingRight": 0,
              },
              Object {
                "color": "#3d3b3b",
                "fontFamily": "Open Sans",
                "fontSize": 14,
                "fontWeight": "300",
                "paddingBottom": 6,
                "paddingLeft": 34,
                "paddingTop": 6,
              },
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          pointerEvents="none"
          style={
            Object {
              "height": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#c1c1c1",
                "height": 1,
                "position": "absolute",
                "width": 0,
              }
            }
          />
          <View
            collapsable={undefined}
            style={
              Object {
                "backgroundColor": "#faa632",
                "height": 1,
                "left": 0,
                "position": "absolute",
                "width": 0,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {},
        Array [
          Object {
            "height": 50,
            "marginTop": 40,
          },
          undefined,
        ],
      ]
    }
  >
    <BVLinearGradient
      colors={
        Array [
          4292519491,
          4292238109,
        ]
      }
      enabled={true}
      endPoint={undefined}
      locations={null}
      onPress={[Function]}
      rounded={true}
      startPoint={undefined}
      style={
        Array [
          Object {
            "borderRadius": 4,
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "opacity": 1,
          },
          undefined,
        ]
      }
    >
      <MKTouchable
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        className={undefined}
        enabled={true}
        hitSlop={undefined}
        maskBorderRadius={2}
        maskColor="rgba(255,255,255,0.15)"
        maskDuration={200}
        maskEnabled={true}
        nativeID={undefined}
        onChange={[Function]}
        onLayout={[Function]}
        onPress={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onTouch={[Function]}
        pointerEvents="box-only"
        rippleColor="rgba(255,255,255,0.2)"
        rippleDuration={200}
        rippleLocation="tapLocation"
        shadowAniEnabled={true}
        shadowColor="black"
        shadowOffset={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        shadowOpacity={0}
        shadowRadius={0}
        style={
          Array [
            Array [
              Array [
                Object {
                  "alignItems": "center",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "center",
                  "opacity": 1,
                },
                undefined,
              ],
              Object {},
            ],
            Object {
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
            },
          ]
        }
        testID={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          pointerEvents="none"
          size="small"
          style={
            Array [
              Object {
                "color": "white",
                "fontFamily": "Open Sans",
                "fontSize": 16,
                "fontWeight": "600",
              },
              undefined,
            ]
          }
        >
          Submit
        </Text>
        <View
          collapsable={undefined}
          style={
            Object {
              "backgroundColor": "rgba(255,255,255,0.15)",
              "borderRadius": 0,
              "height": 1,
              "left": -0,
              "opacity": 0,
              "overflow": "hidden",
              "position": "absolute",
              "top": -0,
              "width": 1,
            }
          }
        >
          <View
            collapsable={undefined}
            style={
              Object {
                "backgroundColor": "rgba(255,255,255,0.2)",
                "borderRadius": 0,
                "height": 0,
                "left": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "scale": 0,
                  },
                ],
                "width": 0,
              }
            }
          />
        </View>
      </MKTouchable>
    </BVLinearGradient>
  </View>
</View>
`;

exports[`<PasswordForm /> light: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {
          "paddingBottom": 0,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 0,
        },
        undefined,
      ],
    ]
  }
>
  <View
    bottom={undefined}
    style={
      Array [
        Object {},
        Array [
          Object {
            "marginBottom": 0,
            "marginTop": 15,
          },
          undefined,
        ],
      ]
    }
    top={undefined}
  >
    <View
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "left": 9,
              "position": "absolute",
              "top": 7,
              "width": 40,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#4b4c4d",
                "fontSize": 16,
              },
              Array [
                Object {
                  "color": "#4b4c4d",
                },
                undefined,
              ],
              Object {
                "fontFamily": "icomoon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "height": 39,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onBlur={[Function]}
          onChange={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          password={true}
          placeholder="Type your password.."
          placeholderTextColor="#3d3b3b"
          secureTextEntry={true}
          selectionColor="#faa632"
          style={
            Array [
              Object {
                "alignSelf": "stretch",
                "backgroundColor": "transparent",
                "marginTop": 0,
                "paddingBottom": 2,
                "paddingTop": 2,
              },
              Object {
                "color": "rgb(117, 117, 117)",
                "fontSize": 16,
                "paddingLeft": 0,
                "paddingRight": 0,
              },
              Object {
                "color": "#3d3b3b",
                "fontFamily": "Open Sans",
                "fontSize": 14,
                "fontWeight": "300",
                "paddingBottom": 6,
                "paddingLeft": 34,
                "paddingTop": 6,
              },
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          pointerEvents="none"
          style={
            Object {
              "height": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#4b4c4d",
                "height": 1,
                "position": "absolute",
                "width": 0,
              }
            }
          />
          <View
            collapsable={undefined}
            style={
              Object {
                "backgroundColor": "#faa632",
                "height": 1,
                "left": 0,
                "position": "absolute",
                "width": 0,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {},
        Array [
          Object {
            "height": 50,
            "marginTop": 40,
          },
          undefined,
        ],
      ]
    }
  >
    <BVLinearGradient
      colors={
        Array [
          4292519491,
          4292238109,
        ]
      }
      enabled={true}
      endPoint={undefined}
      locations={null}
      onPress={[Function]}
      rounded={true}
      startPoint={undefined}
      style={
        Array [
          Object {
            "borderRadius": 4,
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "opacity": 1,
          },
          undefined,
        ]
      }
    >
      <MKTouchable
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        className={undefined}
        enabled={true}
        hitSlop={undefined}
        maskBorderRadius={2}
        maskColor="rgba(255,255,255,0.15)"
        maskDuration={200}
        maskEnabled={true}
        nativeID={undefined}
        onChange={[Function]}
        onLayout={[Function]}
        onPress={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onTouch={[Function]}
        pointerEvents="box-only"
        rippleColor="rgba(255,255,255,0.2)"
        rippleDuration={200}
        rippleLocation="tapLocation"
        shadowAniEnabled={true}
        shadowColor="black"
        shadowOffset={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        shadowOpacity={0}
        shadowRadius={0}
        style={
          Array [
            Array [
              Array [
                Object {
                  "alignItems": "center",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "center",
                  "opacity": 1,
                },
                undefined,
              ],
              Object {},
            ],
            Object {
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
            },
          ]
        }
        testID={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          pointerEvents="none"
          size="small"
          style={
            Array [
              Object {
                "color": "white",
                "fontFamily": "Open Sans",
                "fontSize": 16,
                "fontWeight": "600",
              },
              undefined,
            ]
          }
        >
          Submit
        </Text>
        <View
          collapsable={undefined}
          style={
            Object {
              "backgroundColor": "rgba(255,255,255,0.15)",
              "borderRadius": 0,
              "height": 1,
              "left": -0,
              "opacity": 0,
              "overflow": "hidden",
              "position": "absolute",
              "top": -0,
              "width": 1,
            }
          }
        >
          <View
            collapsable={undefined}
            style={
              Object {
                "backgroundColor": "rgba(255,255,255,0.2)",
                "borderRadius": 0,
                "height": 0,
                "left": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "scale": 0,
                  },
                ],
                "width": 0,
              }
            }
          />
        </View>
      </MKTouchable>
    </BVLinearGradient>
  </View>
</View>
`;

exports[`<PasswordForm /> values dark: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {
          "paddingBottom": 0,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 0,
        },
        undefined,
      ],
    ]
  }
>
  <View
    bottom={undefined}
    style={
      Array [
        Object {},
        Array [
          Object {
            "marginBottom": 0,
            "marginTop": 15,
          },
          undefined,
        ],
      ]
    }
    top={undefined}
  >
    <View
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "left": 9,
              "position": "absolute",
              "top": 7,
              "width": 40,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#c1c1c1",
                "fontSize": 16,
              },
              Array [
                Object {
                  "color": "#c1c1c1",
                },
                undefined,
              ],
              Object {
                "fontFamily": "icomoon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "height": 39,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onBlur={[Function]}
          onChange={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          password={true}
          placeholder="Type your password.."
          placeholderTextColor="#eeeeee"
          secureTextEntry={true}
          selectionColor="#faa632"
          style={
            Array [
              Object {
                "alignSelf": "stretch",
                "backgroundColor": "transparent",
                "marginTop": 0,
                "paddingBottom": 2,
                "paddingTop": 2,
              },
              Object {
                "color": "rgb(117, 117, 117)",
                "fontSize": 16,
                "paddingLeft": 0,
                "paddingRight": 0,
              },
              Object {
                "color": "#3d3b3b",
                "fontFamily": "Open Sans",
                "fontSize": 14,
                "fontWeight": "300",
                "paddingBottom": 6,
                "paddingLeft": 34,
                "paddingTop": 6,
              },
            ]
          }
          underlineColorAndroid="transparent"
          value="testtest"
        />
        <View
          pointerEvents="none"
          style={
            Object {
              "height": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#c1c1c1",
                "height": 1,
                "position": "absolute",
                "width": 0,
              }
            }
          />
          <View
            collapsable={undefined}
            style={
              Object {
                "backgroundColor": "#faa632",
                "height": 1,
                "left": 0,
                "position": "absolute",
                "width": 0,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {},
        Array [
          Object {
            "height": 50,
            "marginTop": 40,
          },
          undefined,
        ],
      ]
    }
  >
    <BVLinearGradient
      colors={
        Array [
          4292519491,
          4292238109,
        ]
      }
      enabled={true}
      endPoint={undefined}
      locations={null}
      onPress={[Function]}
      rounded={true}
      startPoint={undefined}
      style={
        Array [
          Object {
            "borderRadius": 4,
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "opacity": 1,
          },
          undefined,
        ]
      }
    >
      <MKTouchable
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        className={undefined}
        enabled={true}
        hitSlop={undefined}
        maskBorderRadius={2}
        maskColor="rgba(255,255,255,0.15)"
        maskDuration={200}
        maskEnabled={true}
        nativeID={undefined}
        onChange={[Function]}
        onLayout={[Function]}
        onPress={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onTouch={[Function]}
        pointerEvents="box-only"
        rippleColor="rgba(255,255,255,0.2)"
        rippleDuration={200}
        rippleLocation="tapLocation"
        shadowAniEnabled={true}
        shadowColor="black"
        shadowOffset={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        shadowOpacity={0}
        shadowRadius={0}
        style={
          Array [
            Array [
              Array [
                Object {
                  "alignItems": "center",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "center",
                  "opacity": 1,
                },
                undefined,
              ],
              Object {},
            ],
            Object {
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
            },
          ]
        }
        testID={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          pointerEvents="none"
          size="small"
          style={
            Array [
              Object {
                "color": "white",
                "fontFamily": "Open Sans",
                "fontSize": 16,
                "fontWeight": "600",
              },
              undefined,
            ]
          }
        >
          Submit
        </Text>
        <View
          collapsable={undefined}
          style={
            Object {
              "backgroundColor": "rgba(255,255,255,0.15)",
              "borderRadius": 0,
              "height": 1,
              "left": -0,
              "opacity": 0,
              "overflow": "hidden",
              "position": "absolute",
              "top": -0,
              "width": 1,
            }
          }
        >
          <View
            collapsable={undefined}
            style={
              Object {
                "backgroundColor": "rgba(255,255,255,0.2)",
                "borderRadius": 0,
                "height": 0,
                "left": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "scale": 0,
                  },
                ],
                "width": 0,
              }
            }
          />
        </View>
      </MKTouchable>
    </BVLinearGradient>
  </View>
</View>
`;

exports[`<PasswordForm /> values light: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {
          "paddingBottom": 0,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 0,
        },
        undefined,
      ],
    ]
  }
>
  <View
    bottom={undefined}
    style={
      Array [
        Object {},
        Array [
          Object {
            "marginBottom": 0,
            "marginTop": 15,
          },
          undefined,
        ],
      ]
    }
    top={undefined}
  >
    <View
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "left": 9,
              "position": "absolute",
              "top": 7,
              "width": 40,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#4b4c4d",
                "fontSize": 16,
              },
              Array [
                Object {
                  "color": "#4b4c4d",
                },
                undefined,
              ],
              Object {
                "fontFamily": "icomoon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "height": 39,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onBlur={[Function]}
          onChange={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          password={true}
          placeholder="Type your password.."
          placeholderTextColor="#3d3b3b"
          secureTextEntry={true}
          selectionColor="#faa632"
          style={
            Array [
              Object {
                "alignSelf": "stretch",
                "backgroundColor": "transparent",
                "marginTop": 0,
                "paddingBottom": 2,
                "paddingTop": 2,
              },
              Object {
                "color": "rgb(117, 117, 117)",
                "fontSize": 16,
                "paddingLeft": 0,
                "paddingRight": 0,
              },
              Object {
                "color": "#3d3b3b",
                "fontFamily": "Open Sans",
                "fontSize": 14,
                "fontWeight": "300",
                "paddingBottom": 6,
                "paddingLeft": 34,
                "paddingTop": 6,
              },
            ]
          }
          underlineColorAndroid="transparent"
          value="testtest"
        />
        <View
          pointerEvents="none"
          style={
            Object {
              "height": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#4b4c4d",
                "height": 1,
                "position": "absolute",
                "width": 0,
              }
            }
          />
          <View
            collapsable={undefined}
            style={
              Object {
                "backgroundColor": "#faa632",
                "height": 1,
                "left": 0,
                "position": "absolute",
                "width": 0,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {},
        Array [
          Object {
            "height": 50,
            "marginTop": 40,
          },
          undefined,
        ],
      ]
    }
  >
    <BVLinearGradient
      colors={
        Array [
          4292519491,
          4292238109,
        ]
      }
      enabled={true}
      endPoint={undefined}
      locations={null}
      onPress={[Function]}
      rounded={true}
      startPoint={undefined}
      style={
        Array [
          Object {
            "borderRadius": 4,
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "opacity": 1,
          },
          undefined,
        ]
      }
    >
      <MKTouchable
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        className={undefined}
        enabled={true}
        hitSlop={undefined}
        maskBorderRadius={2}
        maskColor="rgba(255,255,255,0.15)"
        maskDuration={200}
        maskEnabled={true}
        nativeID={undefined}
        onChange={[Function]}
        onLayout={[Function]}
        onPress={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onTouch={[Function]}
        pointerEvents="box-only"
        rippleColor="rgba(255,255,255,0.2)"
        rippleDuration={200}
        rippleLocation="tapLocation"
        shadowAniEnabled={true}
        shadowColor="black"
        shadowOffset={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        shadowOpacity={0}
        shadowRadius={0}
        style={
          Array [
            Array [
              Array [
                Object {
                  "alignItems": "center",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "center",
                  "opacity": 1,
                },
                undefined,
              ],
              Object {},
            ],
            Object {
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
            },
          ]
        }
        testID={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          pointerEvents="none"
          size="small"
          style={
            Array [
              Object {
                "color": "white",
                "fontFamily": "Open Sans",
                "fontSize": 16,
                "fontWeight": "600",
              },
              undefined,
            ]
          }
        >
          Submit
        </Text>
        <View
          collapsable={undefined}
          style={
            Object {
              "backgroundColor": "rgba(255,255,255,0.15)",
              "borderRadius": 0,
              "height": 1,
              "left": -0,
              "opacity": 0,
              "overflow": "hidden",
              "position": "absolute",
              "top": -0,
              "width": 1,
            }
          }
        >
          <View
            collapsable={undefined}
            style={
              Object {
                "backgroundColor": "rgba(255,255,255,0.2)",
                "borderRadius": 0,
                "height": 0,
                "left": 0,
                "top": 0,
                "transform": Array [
                  Object {
                    "scale": 0,
                  },
                ],
                "width": 0,
              }
            }
          />
        </View>
      </MKTouchable>
    </BVLinearGradient>
  </View>
</View>
`;
